### PR TITLE
Change the configuration format

### PR DESCRIPTION
**Pull request by Baccata**
1. Make the protocol we're borrowing tests from explicit
2. Make some parameters optional
3. Allow to disallow tests altogether
4. Tweak the names

### CODE REVIEW

File: modules/protocol-tests/resources/META-INF/smithy/test-config.json [hunk 1]
```diff
@@ -1,1592 +1,477 @@
...
+          {
+            "id": "RestJsonHttpRequestWithLabelsAndTimestampFormat"
+          },
+          {
+            "id": "RestJsonHttpRequestLabelEscaping"
+          },
+          {
+            "id": "RestJsonInputAndOutputWithStringHeaders",
+            "appliesTo": "client",
+            "testType": "response"
```
Comment (daddykotex) [inline modules/protocol-tests/resources/META-INF/smithy/test-config.json:310]: I'm assuming if those two aren't specified they fallback to `appliesTo`: both server and client and `testType`  both request and response

Comment (Baccata) [reply modules/protocol-tests/resources/META-INF/smithy/test-config.json:310]: your assumption is correct

**Comment by daddykotex** [comment]
Should we bump minor when we release this?

**Comment by Baccata** [comment]
> Should we bump minor when we release this?

@daddykotex not sure tbh. I'll do it to be safe.

**Pull request merged**